libs/langchain/langchain/document_loaders/recursive_url_loader.py

```diff
@@ -132,53 +92,45 @@ def _get_child_links_recursive(
         Args:
             url: The URL to crawl.
             visited: A set of visited URLs.
+            depth: Current depth of recursion. Stop when depth >= max_depth.
         """
 
-        if depth > self.max_depth:
-            return []
-
-        # Add a trailing slash if not present
-        if not url.endswith("/"):
-            url += "/"
+        if depth >= self.max_depth:
+            return
+        # Exclude the links that start with any of the excluded directories
+        if any(url.startswith(exclude_dir) for exclude_dir in self.exclude_dirs):
+            return
 
         # Exclude the root and parent from a list
         visited = set() if visited is None else visited
 
-        # Exclude the links that start with any of the excluded directories
-        if self.exclude_dirs and any(
-            url.startswith(exclude_dir) for exclude_dir in self.exclude_dirs
-        ):
-            return []
-
         # Get all links that can be accessed from the current URL
         try:
             response = requests.get(url, timeout=self.timeout)
         except Exception:
-            return []
-
-        absolute_paths = self._get_sub_links(response.text, url)
+            logger.warning(f"Unable to load from {url}")
+            return
+        content = self.extractor(response.text)
+        if content:
+            yield Document(
```

Comment:
note: yield current document instead of children (meaning init root doc included)